src/index.ts

```diff
@@ -1,62 +1,61 @@
+        container?: interfaces.Container;
+        dependencies?: { [key: string]: interfaces.ServiceIdentifier<unknown> };
     }
 }
 
-type InjectableVueClass = VueConstructor<Vue> & {
-    options: ComponentOptions<Vue>
-};
-
 const VueInversify: PluginFunction<unknown> = instance => {
     instance.mixin({
         beforeCreate() {
             const dependencies = this.$options.dependencies;
-            const container = this.$options.container || (this.$options.container = getContainer(this));
             
             if (dependencies) {
+                const container = this.$options.container || (this.$options.container = getContainer(this));
+
                 if (!container)
                     throw new TypeError('vue-inversify: encountered props marked with @inject but no container was provided when creating the Vue instance');
-
+                
                 Object.keys(dependencies).forEach(key => {
+                    if (Object.getOwnPropertyDescriptor(this, key))
+                        return;
+
                     Object.defineProperty(this, key, {
                         enumerable: true,
                         value: container.get(dependencies[key])
                     });
-                });
+                });            
             }
         }
     });
 };
 
 const inject = (identifier?: interfaces.ServiceIdentifier<unknown>) => {
     return function(target: Vue, prop: string) {
-        // Try use reflection to generate the identifier if not provided explicitly.
-        if (!identifier && typeof(Reflect) !== 'undefined' && typeof(Reflect.getMetadata) === 'function')
-            identifier = Reflect.getMetadata('design:type', target, prop) as interfaces.ServiceIdentifier<unknown>;
-
-        if (!identifier)
-            throw new TypeError('Unable to resolve identifier; please specify it explicitly');
+        createDecorator((options, key) => {
+            // Try use reflection to generate the identifier if not provided explicitly.
+            if (!identifier && typeof(Reflect) !== 'undefined' && typeof(Reflect.getMetadata) === 'function')
+                identifier = Reflect.getMetadata('design:type', target, prop) as interfaces.ServiceIdentifier<unknown>;
 
-        const injectable = (typeof(target) === 'function' ? target : target.constructor) as InjectableVueClass;
-        const options = injectable.options || (injectable.options = {});
-        const dependencies = options.dependencies || (options.dependencies = {});
+            if (!identifier)
+                throw new TypeError('vue-inversify: Unable to resolve identifier; please specify it explicitly');
 
-        dependencies[prop] = identifier;
+            const dependencies = options.dependencies || (options.dependencies = {});
+            dependencies[key] = identifier;
+        })(target, prop);
     }
 };
 
-function getContainer(component: Vue): interfaces.Container | undefined {
```

Comment:
hmm, I can't remember why I would have either added it in the first place outside of overriding types while testing, but also this would be 1. internal only and 2. the only one explicit return types anyway